iree/tools/BUILD

```diff
@@ -225,15 +225,6 @@ cc_binary(
     ],
 )
 
-cc_binary(
-    name = "iree-translate",
```

Comment:
this (and the CMake) will be removed from `tools/BUILD`, not `iree/tools/BUILD` after https://github.com/google/iree/pull/9135